x-pack/plugins/uptime/public/components/overview/filter_group/filter_group.tsx

```diff
@@ -5,100 +5,76 @@
 
-  const { selectedLocations, selectedPorts, selectedSchemes, selectedTags } = useSelectedFilters();
+  const { filtersList } = useSelectedFilters();
 
-  const onFilterFieldChange = (fieldName: string, values: string[]) => {
-    setUpdatedFieldValues({ fieldName, values });
-  };
+  const indexPattern = useIndexPattern();
 
-  const isMonitorPage = useRouteMatch(MONITOR_ROUTE);
+  const [isOpen, setIsOpen] = useState('');
 
-  const filterPopoverProps: FilterPopoverProps[] = [
-    {
-      loading,
-      onFilterFieldChange,
-      fieldName: 'observer.geo.name',
-      id: 'location',
-      items: locations || [],
-      selectedItems: selectedLocations,
-      title: filterLabels.LOCATION,
+  const onFilterFieldChange = useCallback(
+    (fieldName: string, values?: string[], notValues?: string[]) => {
+      setUpdatedFieldValues({ fieldName, values, notValues });
+      setIsOpen('');
     },
-    // on monitor page we only display location filter in ping list
-    ...(!isMonitorPage
-      ? [
-          {
-            loading,
-            onFilterFieldChange,
-            fieldName: 'url.port',
-            id: 'port',
-            disabled: isDisabled(ports),
-            items: ports?.map((p: number) => p.toString()) ?? [],
-            selectedItems: selectedPorts,
-            title: filterLabels.PORT,
-          },
-          {
-            loading,
-            onFilterFieldChange,
-            fieldName: 'monitor.type',
-            id: 'scheme',
-            disabled: isDisabled(schemes),
-            items: schemes ?? [],
-            selectedItems: selectedSchemes,
-            title: filterLabels.SCHEME,
-          },
-          {
-            loading,
-            onFilterFieldChange,
-            fieldName: 'tags',
-            id: 'tags',
-            disabled: isDisabled(tags),
-            items: tags ?? [],
-            selectedItems: selectedTags,
-            title: filterLabels.TAG,
-          },
-        ]
-      : []),
-  ];
+    []
+  );
 
   return (
-    <Container>
-      {filterPopoverProps.map((item) => (
-        <FilterPopover key={item.id} {...item} />
-      ))}
-    </Container>
+    <>
+      <Container>
+        {indexPattern &&
+          filtersList.map(({ field, label, selectedItems, excludedItems }) => (
+            <FieldValueSuggestions
+              key={field}
+              compressed={false}
+              indexPatternTitle={indexPattern.title}
+              sourceField={field}
+              label={label}
+              selectedValue={selectedItems}
+              excludedValue={excludedItems}
+              onChange={(values, notValues) => onFilterFieldChange(field, values, notValues)}
+              asCombobox={false}
+              asFilterButton={true}
+              forceOpen={isOpen === field}
+              setForceOpen={() => {
+                setIsOpen('');
+              }}
+              filters={[]}
+              isSyntheticsData={true}
```

Comment:
What are your thoughts on making this more generic and specifying the name of the field to use for cardinality? In other words, the option could be `count="monitor.id"`

Reply:
that's a great idea actually, i am handing over this PR to @dominiqueclarke in case she has capacity to pick this up. She can probably address this suggestion.